### ISSUE TITLE
RFC: removing docker build

### ISSUE THREAD

**Issue by danobi** [iovisor/bpftrace#332]
Purely exploratory issue. Is anyone using the docker build?

From what I can tell, distros prefer packaging without docker. Docker might be more convenient for newcomers to the project, but we're essentially supporting dual build systems. If we remove docker support, it could unify our efforts and make everyone happier in the long term.

**Comment by mmarchini** [comment]
Out of curiosity, what would change if we remove docker builds? I don't remember anything specific to docker builds in our CMakeList files.

Sometimes I use docker to build bpftrace with different llvm versions, but I wrote my own images. I think @williangaspar and @ajor are using `./build.sh`.

**Comment by danobi** [comment]
> Out of curiosity, what would change if we remove docker builds?

After looking at it more closely, I don't think much.

> Sometimes I use docker to build bpftrace with different llvm versions, but I wrote my own images

Oh yeah that does sound like a good use case.

**Comment by brendangregg** [comment]
FWIW, I've never used the docker builds.

**Comment by famz** [comment]
What was the consensus of this one? I am new to bpftrace and find it convenient to build a statically linked bpftrace binary which is portable. Though now I have a syntax error on master:

```
[ 20%] Building CXX object src/CMakeFiles/bpftrace.dir/clang_parser.cpp.o
/data00/fam/bpftrace/src/clang_parser.cpp: In member function 'void bpftrace::ClangParser::parse(bpftrace::ast::Program*, bpftrace::StructMap&)':
/data00/fam/bpftrace/src/clang_parser.cpp:223:8: error: expected unqualified-id before '[' token
   auto [ksrc, kobj] = get_kernel_dirs(utsname);
        ^
/data00/fam/bpftrace/src/clang_parser.cpp:224:7: error: 'ksrc' was not declared in this scope
   if (ksrc != "")
       ^~~~
/data00/fam/bpftrace/src/clang_parser.cpp:225:55: error: 'kobj' was not declared in this scope
     kflags = get_kernel_cflags(utsname.machine, ksrc, kobj);
                                                       ^~~~
```

What is our supported C++ version? Should I be looking at fixing the docker image or the source code? Should I look for a more proper setup to build bpftrace?

**Comment by williangaspar** [comment]
@famz I had the same problem. I've updated my gcc and c++ to version 8.

**Comment by mmarchini** [comment]
@famz we just landed #423 and #433, which should fix the issues when building statically on the Alpine container. Please note that there are some known issues with the static build binary though: https://github.com/iovisor/bpftrace/issues/266
